### ISSUE TITLE
Kosár kedvezmény nem megy át a simple felületre

### ISSUE THREAD

**Issue by jofbe** [conedevelopment/simplepay-gateway#106]
- Simplepay Gateway bővítmény verzió: 2.5.3
- WooCommerce verzió: 6.9.3
- WordPress verzió: 6.0.3
- PHP verzió: 7.4.25

Sziasztok!
Az egyik oldalon használjuk a Woo Discount Rules plugin-t (v.: 2.4.4), ami úgy van beállítva, hogy x pénz elköltése után a vásárló törzsvásárló lesz, és 10% kerdvezményt kap (percentage cart adjustment). Az egyik ilyen vásárló a legutóbbi tranzakciónál jelezte, hogy a kosárban, a pénztár oldalon, és a számlázz.hu-s számlán is rajta van a kedvezmény, de a Simple oldalán a kedvezmény nélküli árat kellett fizetnie. Korábban nem volt ilyen problémánk.

`IPNs
OK, Finish, 2022.10.20 19:18:06`
`Sent emails
OK, Customer finish, 2022.10.20 19:18:07
OK, Merchant finish, 2022.10.20 19:18:07`

`{"salt":"Xi1eX9iRmJ8NApEi0xxqdnk3Y3A9tQjd","orderRef":"wc-12424","method":"CARD","merchant":"__________","finishDate":"2022-10-20T19:18:01+02:00","paymentDate":"2022-10-20T19:16:18+02:00","transactionId":318132261,"status":"FINISHED"}`

Hol mászhatott el a dolog?

**Comment by iamgergo** [comment]
Szia!

Erről itt írunk bővebben: https://simplepay.conedevelopment.com/a-simplepay-bovitmeny-korlatai, a kedvezmények résznél.

Tehát a SimplePay felületén fizetendő összeg eltért a számlán szereplő összegtől?

**Comment by jofbe** [comment]
Szia!
Igen, eltér a számlán levő végösszeg és a fizetett összeg, a kedvezmény mértékével.

Azóta kiderült, hogy 1 vásárlás kivételével mindegyik törzsvásárló többet fizetett, mint kellett volna, ami rendkívül kellemetlen a bolt tulajdonosnak.

![image](https://user-images.githubusercontent.com/2345809/197480296-ddb3f719-1aee-4c18-b89a-4e8e115dd610.png)

![image](https://user-images.githubusercontent.com/2345809/197480249-55f01b85-6839-4a51-83ac-eefd9d621108.png)

**Comment by iamgergo** [comment]
Szia!

Ez valóban kellemetlen, de ez sajnos benne van, nem tudjuk felkészíteni a bővítményt hogy minden más bővítménnyel kompatibilis legyen.

Mi volt más abban az egy vásárlásban, azt lehet tudni?

Köszi!

**Comment by iamgergo** [comment]
@jofbe Itt még annyi kérdésem lenne, hogy ez esetben a fee az negatív?

**Comment by jofbe** [comment]
Mind a kettő esetben negatív számként jelenik meg a kedvezmény.
Kupon:
![image](https://user-images.githubusercontent.com/2345809/197734635-360c1263-4311-4d8a-9469-e05515270ea1.png)
_order_total: a kedvezmény már benne van, ennyi is lett fizetve
_cart_discount: az eredeti rendelés 10%-a

Fee:
![image](https://user-images.githubusercontent.com/2345809/197734965-9dcc757b-59f0-45e9-9c0e-5896cc8d39d9.png)
_order_total: a kedvezmény már benne van, mégsem ennyi lett a simple felületen levonva
_cart_discount: 0

**Comment by iamgergo** [comment]
Igen, tehát a gond itt az, hogy a SimplePay-nek nem lehet átadni negatív értékű elemet. Ezért ezt már az átadás előtt ki kell szűrni, egyébként hibát ad a tranzakció.

A negatív fee-t pedig nem lehet egy az egyben átadni mint discount, mert az egyéb gondokat okozna. Így sajnos az az egyetlen járható út jelen esetben, amit te is említettél fent.

**Comment by jofbe** [comment]
Abban tudsz segíteni, hogy a simple plugin az adatbázisban milyen helyeken (tábla+kulcs) keresi az esetleges kedvezményeket? Megnézném, hogy ott negatív számként van-e tárolva.

**Comment by iamgergo** [comment]
A plugin nem közvetlenül DB-ből keresi. A Woo API-ban van ilyen metódus amivel az adott kosár/rendelés elemeit lekérdezzük: https://github.com/conedevelopment/simplepay-gateway/blob/master/src/Payloads/PaymentPayload.php#L112-L124
